tensorflow_addons/optimizers/conditional_gradient_test.py

```diff
@@ -67,109 +67,141 @@ def test_like_dist_belief_nuclear_cg01():
+    )
+    _ = cg_opt.apply_gradients(zip([grads0, grads1], [var0, var1]))
+
+    # Check we have slots
+    assert ["conditional_gradient"] == cg_opt.get_slot_names()
+    slot0 = cg_opt.get_slot(var0, "conditional_gradient")
+    assert slot0.get_shape() == var0.get_shape()
+    slot1 = cg_opt.get_slot(var1, "conditional_gradient")
+    assert slot1.get_shape() == var1.get_shape()
+
+    test_utils.assert_allclose_according_to_type(
+        np.array(
+            [
+                1.0 * 0.5 - (1 - 0.5) * 0.01 * 0.1 / norm0,
+                2.0 * 0.5 - (1 - 0.5) * 0.01 * 0.1 / norm0,
+            ]
+        ),
+        var0.numpy(),
+    )
+    test_utils.assert_allclose_according_to_type(
+        np.array(
+            [
+                3.0 * 0.5 - (1 - 0.5) * 0.01 * 0.01 / norm1,
+                4.0 * 0.5 - (1 - 0.5) * 0.01 * 0.01 / norm1,
+            ]
+        ),
+        var1.numpy(),
+    )
+
+    # Step 2: the conditional_gradient contain the previous update.
+    cg_opt.apply_gradients(zip([grads0, grads1], [var0, var1]))
+    test_utils.assert_allclose_according_to_type(
+        np.array(
+            [
+                (1.0 * 0.5 - (1 - 0.5) * 0.01 * 0.1 / norm0) * 0.5
+                - (1 - 0.5) * 0.01 * 0.1 / norm0,
+                (2.0 * 0.5 - (1 - 0.5) * 0.01 * 0.1 / norm0) * 0.5
+                - (1 - 0.5) * 0.01 * 0.1 / norm0,
+            ]
+        ),
+        var0.numpy(),
+    )
+    test_utils.assert_allclose_according_to_type(
+        np.array(
+            [
+                (3.0 * 0.5 - (1 - 0.5) * 0.01 * 0.01 / norm1) * 0.5
+                - (1 - 0.5) * 0.01 * 0.01 / norm1,
+                (4.0 * 0.5 - (1 - 0.5) * 0.01 * 0.01 / norm1) * 0.5
+                - (1 - 0.5) * 0.01 * 0.01 / norm1,
+            ]
+        ),
+        var1.numpy(),
+    )
 
-            # Check that the parameters have been updated.
-            norm0 = self.evaluate(norm0)
-            norm1 = self.evaluate(norm1)
 
-            self.assertAllCloseAccordingToType(
-                np.array(
-                    [
-                        1.0 * 0.5 - (1 - 0.5) * 0.01 * 0.1 / norm0,
-                        2.0 * 0.5 - (1 - 0.5) * 0.01 * 0.1 / norm0,
-                    ]
-                ),
-                self.evaluate(var0),
-            )
-            self.assertAllCloseAccordingToType(
-                np.array(
-                    [
-                        3.0 * 0.5 - (1 - 0.5) * 0.01 * 0.01 / norm1,
-                        4.0 * 0.5 - (1 - 0.5) * 0.01 * 0.01 / norm1,
-                    ]
-                ),
-                self.evaluate(var1),
-            )
+@pytest.mark.parametrize("dtype", [(tf.half, 0), (tf.float32, 1), (tf.float64, 2)])
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("use_resource", [True, False])
+def test_basic_callable_params_frobenius(dtype, use_resource):
+    do_test_basic_frobenius(dtype, use_resource, use_callable_params=True)
```

Comment:
There is no need for a second function `test_basic_callable_params_frobenius` here.

You can rename `do_test_basic_frobenius` to `test_basic_frobenius` and make it the main test.